File: src/main/java/com/globaldefensesurveillancetracking/service/SoldierService.java
package com.globaldefensesurveillancetracking.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.globaldefensesurveillancetracking.domain.Soldier;
import com.globaldefensesurveillancetracking.dto.SoldierDTO;
import com.globaldefensesurveillancetracking.dto.SoldierSearchDTO;
import com.globaldefensesurveillancetracking.dto.SoldierPageDTO;
import com.globaldefensesurveillancetracking.dto.SoldierConvertCriteriaDTO;
import com.globaldefensesurveillancetracking.service.GenericService;
import com.globaldefensesurveillancetracking.dto.common.RequestDTO;
import com.globaldefensesurveillancetracking.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SoldierService extends GenericService<Soldier, Integer> {

	List<Soldier> findAll();

	ResultDTO addSoldier(SoldierDTO soldierDTO, RequestDTO requestDTO);

	ResultDTO updateSoldier(SoldierDTO soldierDTO, RequestDTO requestDTO);

    Page<Soldier> getAllSoldiers(Pageable pageable);

    Page<Soldier> getAllSoldiers(Specification<Soldier> spec, Pageable pageable);

	ResponseEntity<SoldierPageDTO> getSoldiers(SoldierSearchDTO soldierSearchDTO);
	
	List<SoldierDTO> convertSoldiersToSoldierDTOs(List<Soldier> soldiers, SoldierConvertCriteriaDTO convertCriteria);

	SoldierDTO getSoldierDTOById(Integer soldierId);







}

File: globaldefensesurveillancetracking_ui/src/services/SoldierService.js
import http from "../http-common";

class SoldierService {
  getAllSoldiers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/soldier/soldiers`, searchDTO);
  }

  get(soldierId) {
    return this.getRequest(`/soldier/${soldierId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/soldier?field=${matchData}`, null);
  }

  addSoldier(data) {
    return http.post("/soldier/addSoldier", data);
  }

  update(data) {
  	return http.post("/soldier/updateSoldier", data);
  }
  
  uploadImage(data,soldierId) {
  	return http.postForm("/soldier/uploadImage/"+soldierId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SoldierService();
